### PR TITLE
save junint xml output in the workspace

### PR DESCRIPTION
**Pull request by MaximilianMeister**
Signed-off-by: Maximilian Meister <mmeister@suse.de>

## DEPENDS ON
https://github.com/kubic-project/automation/pull/12

### CODE REVIEW

File: vars/bootstrapEnvironment.groovy [hunk 1]
```diff
@@ -32,7 +32,7 @@ Environment call(Map parameters = [:]) {
                 // TODO: drop this after switching to a VM based admin setup
                 "DEVENV=true",
             ]) {
-                sh(script: 'bundle exec rspec spec/**/*')
+                sh(script: "bundle exec --format RspecJunitFormatter --out ${WORKSPACE}/velum-bootstrap.xml rspec spec/**/*")
```
Comment (MaximilianMeister) [inline vars/bootstrapEnvironment.groovy]: is this output path correct? `${WORKSPACE}/velum-bootstrap.xml` ?

Comment (inercia) [inline vars/bootstrapEnvironment.groovy]: Where should the junit xml files go? `${WORKSPACE}`? `${WORKSPACE}/logs`?

Comment (kiall) [inline vars/bootstrapEnvironment.groovy]: I would not have a gatherJunit step, as Jenkins treats the absence of an expected junit file as "Wow, something went REALLY wrong"... When these are reused, kube-e2e etc won't always be ran.

We should trigger the junit step right after we run the tests that produce it :)

Comment (kiall) [inline vars/bootstrapEnvironment.groovy]: > bundler: command not found: --format

Comment (kiall) [inline vars/bootstrapEnvironment.groovy]: We should also read the junit file in :)

    junit "${WORKSPACE}/velum-bootstrap.xml"

**Comment by MaximilianMeister** [comment]
> bundler: command not found: --format

that's the missing dependent PR mentioned above: https://github.com/kubic-project/automation/pull/12

> We should also read the junit file in

so it's enough to call that right after the script? i don't want to influence the test result with another command afterwards, not sure how that works with `junit`

File: vars/bootstrapEnvironment.groovy [hunk 2]
```diff
@@ -32,7 +32,11 @@ Environment call(Map parameters = [:]) {
                 // TODO: drop this after switching to a VM based admin setup
                 "DEVENV=true",
             ]) {
-                sh(script: 'bundle exec rspec spec/**/*')
+                try {
+                    sh(script: "bundle exec --format RspecJunitFormatter --out ${WORKSPACE}/velum-bootstrap.xml rspec spec/**/*")
+                } finally {
+                    junit "${WORKSPACE}/velum-bootstrap.xml"
```
Comment (MaximilianMeister) [inline vars/bootstrapEnvironment.groovy]: @kiall is this correct? i think if i wouldn't use a `finally` here we wouldn't reach the junit call if the sh test failed, or?

File: vars/bootstrapEnvironment.groovy [hunk 3]
```diff
@@ -32,7 +32,11 @@ Environment call(Map parameters = [:]) {
                 // TODO: drop this after switching to a VM based admin setup
                 "DEVENV=true",
             ]) {
-                sh(script: 'bundle exec rspec spec/**/*')
+                try {
+                    sh(script: "bundle exec rspec --format RspecJunitFormatter --out ${WORKSPACE}/velum-bootstrap.xml spec/**/*")
+                } finally {
+                    junit "${WORKSPACE}/velum-bootstrap.xml"
```
Comment (MaximilianMeister) [inline vars/bootstrapEnvironment.groovy]: somehow this doesn't work, it can't find the file, any ideas what could be wrong here?

Comment (kiall) [inline vars/bootstrapEnvironment.groovy]: For some reason, it won't accept an absolute path - likely Jenkin's fault. You'll need to change to the WORKSPACE dir, and `junit("velum-bootstrap.xml")`

**Pull request merged**